client/request.go

```diff
@@ -236,16 +236,14 @@ func (cli *Client) addHeaders(req *http.Request, headers headers) *http.Request
 	// Add CLI Config's HTTP Headers BEFORE we set the Docker headers
 	// then the user can't change OUR headers
 	for k, v := range cli.customHTTPHeaders {
-		if versions.LessThan(cli.version, "1.25") && k == "User-Agent" {
+		if versions.LessThan(cli.version, "1.25") && http.CanonicalHeaderKey(k) == "User-Agent" {
```

Comment:
FWIW, not entirely sure why we're ignoring User-Agent only for older clients. This was part of the version negotiation added in https://github.com/moby/moby/commit/e98e4a71110fd33852bb755a9b8b4ebc9df904db (https://github.com/moby/moby/pull/27745)

Would have to do more digging, probably could use a comment added here to describe what/why